src/thread.rs

```diff
@@ -1,5 +1,11 @@
 use std::fmt::{self, Debug};
-use std::thread::{self, ThreadId};
+use std::sync::atomic::Ordering;
+
+#[cfg(not(syn_no_atomic_u64))]
+use std::{num::NonZeroU64 as NonZeroThreadId, sync::atomic::AtomicU64 as AtomicThreadId};
```

Comment:
Beware, some 32-bit targets do not have `Atomicu64`!
file:///usr/share/doc/rust/html/std/sync/atomic/index.html#portability

Reply:
I don't anticipate having to land this. :wink: